scripts/protocgen.sh

```diff
@@ -27,7 +27,20 @@ Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
   -I "$cosmos_sdk_dir/proto" \
   --grpc-gateway_out=logtostderr=true:. \
   $(find "${dir}" -maxdepth 1 -name '*.proto')
+  # command to generate docs using protoc-gen-doc
 done
 
+
+# command to generate docs using protoc-gen-doc
+buf protoc \
+  -I "proto" \
+  -I "$cosmos_sdk_dir/third_party/proto" \
+  -I "$cosmos_sdk_dir/proto" \
+  --doc_out=./docs/core \
+  --doc_opt=./docs/protodoc-markdown.tmpl,proto-docs.md \
+  $(find "$(pwd)/proto" -maxdepth 4 -name '*.proto')
+go mod tidy
```

Comment:
Why do we tidy here?

In any case, we should ensure this has the 1.17 compat:

```
$ go mod tidy -compat=1.17
```

Reply:
actually no need to tidy here!